### PR TITLE
Improved FAQ readability with collapsible Q&A.

### PR DESCRIPTION
**Pull request by lucaspar**
### Changes

+ Changed Q&A from plain text to collapsible sections in `README.md`.
+ Fixed typo in FAQ answer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,49 +175,77 @@ There are several data loading implementations of popular datasets across differ
...
+You can
 [implement your own custom DataPipe](https://pytorch.org/data/main/tutorial.html#implementing-a-custom-datapipe). If you
 believe your use case is common enough such that the community can benefit from having your custom DataPipe added to
 this library, feel free to open a GitHub issue. We will be happy to discuss!
+</details>
 
-Q: What happens when the `Shuffler` DataPipe is used with DataLoader?
+<details>
+<summary>
+What happens when the <code>Shuffler</code> DataPipe is used with DataLoader?
```
Comment (ejguan) [inline README.md:191]: Do we need to use `<code>` here? I think "`" should be sufficient.

File: README.md [hunk 2]
```diff
@@ -175,49 +175,77 @@ There are several data loading implementations of popular datasets across differ
...
+<summary>
+How is multiprocessing handled with DataPipes?
+</summary>
 
-A: Multi-process data loading is still handled by the `DataLoader`, see the
+Multi-process data loading is still handled by the `DataLoader`, see the
 [DataLoader documentation for more details](https://pytorch.org/docs/stable/data.html#single-and-multi-process-data-loading).
 As of PyTorch version >= 1.12.0 (TorchData version >= 0.4.0), data sharding is automatically done for DataPipes within
-the `DataLoader` as long as a `ShardingFiler` DataPipe exists in your pipeline. Please see the
+the `DataLoader` as long as a `ShardingFilter` DataPipe exists in your pipeline. Please see the
```
Comment (ejguan) [inline README.md:225]: Good catch.

**Review by ejguan — COMMENTED** [review]
Thank you. Overall look good to me with one nit comment.

File: README.md (hunk 1 above)
Comment (lucaspar) [reply README.md:191]: Yes - I wasn't going to change the backticks, but if you check the preview, backticks do not render as <code> does between summary tags, unless we add a new line after <summary>.

The problem with that is that the new line is rendered and doesn't look good, so <code> was necessary to make the font monospace.

Comment (ejguan) [reply README.md:191]: I see. Makes sense.

File: README.md [hunk 3]
```diff
@@ -175,49 +175,77 @@ There are several data loading implementations of popular datasets across differ
...
+</details>
 
-Q: What happens when the `Batcher` DataPipe is used with DataLoader?
+<details>
+<summary>
+What happens when the <code>Batcher</code> DataPipe is used with DataLoader?
+</summary>
 
-A: If you choose to use `Batcher` while setting `batch_size > 1` for DataLoader, your samples will be batched more than
+If you choose to use <code>Batcher</code> while setting <code>batch_size > 1</code> for DataLoader, your samples will be batched more than
```
Comment (ejguan) [inline README.md:203]: If this only applies to summary, let's change this part. Then this PR is good to go.

Comment (lucaspar) [reply README.md]: Good call, new commit added.

File: README.md [hunk 4]
```diff
@@ -175,49 +175,77 @@ There are several data loading implementations of popular datasets across differ
...
+</details>
 
-Q: What happens when the `Shuffler` DataPipe is used with DataLoader?
+<details>
+<summary>
+What happens when the <code>Shuffler</code> DataPipe is used with DataLoader?
+</summary>
 
-A. In order to enable shuffling, you need to add a `Shuffler` to your DataPipe line. Then, by default, shuffling will
+In order to enable shuffling, you need to add a <code>Shuffler</code> to your DataPipe line. Then, by default, shuffling will
```
Comment (ejguan) [inline README.md:194]: Sorry. There is another case.

**Comment by ejguan** [comment]
For the lint Error, could you please fix it by following the instruction [here](https://github.com/pytorch/data/blob/main/CONTRIBUTING.md#code-style)?

**Comment by lucaspar** [comment]
It's passing locally, so I think it looks good if anyone can approve running the workflows.

**Pull request closed without merging**